src/DependencyInjection/ProviderFactory.php

```diff
@@ -39,16 +40,19 @@ public function __construct(UrlGeneratorInterface $generator)
      * @param array $options
      * @param string $redirectUri
      * @param array $redirectParams
+     * @param ClientInterface|null $httpClient
      * @return mixed
      */
-    public function createProvider($class, array $options, $redirectUri, array $redirectParams = [])
+    public function createProvider($class, array $options, $redirectUri, array $redirectParams = [], ClientInterface $httpClient = null)
```

Comment:
Since there are other potential `collaborators` keys, should we make this new, 4th argument an array called `$collaborators`? And we would, for this PR, set the `ClientInterface` on the `httpClient` key? I'm worried that if we add more collaborator options later, it will get a bit weird to have a 5th & 6th argument, but they're all optional.